### PR TITLE
Update Airflow version to 2.5.1

### PR DESCRIPTION
**Pull request by pierrejeambrun**
Update airflow version for main following the 2.5.1 release.

### CODE REVIEW

File: RELEASE_NOTES.rst [hunk 1]
```diff
@@ -21,6 +21,119 @@
 
 .. towncrier release notes start
 
+Airflow 2.5.1 (2023-01-20)
+--------------------------
+
+Significant Changes
+^^^^^^^^^^^^^^^^^^^
+
+Trigger gevent ``monkeypatching`` via environment variable (#28283)
```
Comment (jedcunningham) [inline RELEASE_NOTES.rst:30]: You also should delete the newsfragment for this one.

**Review by potiuk — APPROVED** [review]

**Review by jedcunningham — CHANGES_REQUESTED** [review]
If you do the chart here, you also need to adjust that newsfragment as well.

File: RELEASE_NOTES.rst (hunk 1 above)
Comment (pierrejeambrun) [reply RELEASE_NOTES.rst:30]: Good catch Jed, I missed that, thanks. I looked for a `28283` newsfragment, but that is the PR number not the ticket :facepalm: `08212`.

**Comment by pierrejeambrun** [comment]
> If you do the chart here, you also need to adjust that newsfragment as well.

Thanks I didn't know that. (adding this to my [small release process update](https://github.com/apache/airflow/pull/29098) as well so it's not forgotten in the future).

**Review by jedcunningham — APPROVED** [review]

**Review by ephraimbuddy — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**